### PR TITLE
[harfbuzz] Update library to 2.6.2

### PR DESCRIPTION
**Pull request by ehsan-mohammadi**
- Update `harfbuzz` library to 2.6.2 version.
- ~~Remove `find-package-freetype-2.patch` patch.~~
-  ~~Remove `glib-cmake.patch` patch.~~

### CODE REVIEW

**Comment by ehsan-mohammadi** [comment]
@cbezault I updated the `harfbuzz` and it successfully installed on my system. But in this PR, I get `##[error] REGRESSION` error. Can you help me?
Thanks

**Comment by cenit** [comment]
Try to install qt5-base if logs downloaded from CI (build artifacts) are not clear enough for you.
The message means that with the updated harfbuzz qt5 cannot build anymore, so it might require a fix before this update

**Comment by cenit** [comment]
Also please document in this pr why the patches are not needed anymore (accepted upstream, etc.)
In particular, I am skeptic about the find_package_freetype_2 one, I think it might still be needed

**Comment by ehsan-mohammadi** [comment]
@cenit Many thanks for helping me. I'll do your suggestions.

**Comment by ehsan-mohammadi** [comment]
> Also please document in this pr why the patches are not needed anymore (accepted upstream, etc.)
> In particular, I am skeptic about the find_package_freetype_2 one, I think it might still be needed

So sorry, I did a foolish mistake. Just think these patches added to the main harfbuzz `cmakelist.txt` file.

File: ports/harfbuzz/portfile.cmake [hunk 1]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (NancyLi1013) [inline ports/harfbuzz/portfile.cmake]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove this line?

File: ports/harfbuzz/portfile.cmake [hunk 2]
```diff
@@ -16,16 +16,6 @@ vcpkg_from_github(
...
-    message(FATAL_ERROR "Harfbuzz's cmake must not directly include() FindFreetype.")
-endif()
-
-if("${_contents}" MATCHES "find_library\\(GLIB_LIBRARIES")
-    message(FATAL_ERROR "Harfbuzz's cmake must not directly find_library() glib.")
-endif()
-
 SET(HB_HAVE_ICU "OFF")
 if("icu" IN_LIST FEATURES)
     SET(HB_HAVE_ICU "ON")
```
Comment (NancyLi1013) [inline ports/harfbuzz/portfile.cmake]: Could you please use `vcpkg_check_features() `to handle these features?

Comment (NancyLi1013) [reply ports/harfbuzz/portfile.cmake]: Could you update 
`file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/harfbuzz RENAME copyright)
`
as 
`file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`?

Comment (NancyLi1013) [reply ports/harfbuzz/portfile.cmake]: `vcpkg_test_cmake(PACKAGE_NAME harfbuzz)` is deprecated.
Could you remove this line?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/harfbuzz/portfile.cmake (hunk 2 above)
Comment (ehsan-mohammadi) [reply ports/harfbuzz/portfile.cmake]: @NancyLi1013 Thanks for helping me! I resolve these conversations

Comment (ehsan-mohammadi) [reply ports/harfbuzz/portfile.cmake]: > Could you please use `vcpkg_check_features() `to handle these features?

I don't know I used `vcpkg_check_features()` correctly or not. Can you review it again?

**Comment by NancyLi1013** [comment]
It failed on Windows and Linux platform.
Could you please take a look and try to fix the regressions?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @ehsan-mohammadi 
Could you please address the review suggestions and regressions?

**Comment by PhoebeHui** [comment]
Related to https://github.com/microsoft/vcpkg/pull/10146

**Comment by NancyLi1013** [comment]
Ping @ehsan-mohammadi 
Is work still being done for this PR?

**Comment by NancyLi1013** [comment]
@ehsan-mohammadi 
Could you please resolve the conflicts?

**Comment by NancyLi1013** [comment]
@ehsan-mohammadi 
Thanks for the PR; we're closing this for now since there's been no response. If you'd like to continue working on it, please reopen and ping us!

**Pull request closed without merging**